### PR TITLE
docs: make |lspconfig-servers| jump to the markdown file

### PR DESCRIPTION
**Pull request by muniter**
By generating the tags at docgen and modifying the `|lspconfig-servers|` tag we can make `:help lspconfig-servers` take you to `./doc/server_configurations.md` and by adding a modeline to the markdown file we make sure it sets the correct filetype.

@mjlbach Let me know if you like this approach or you would like something different.

### CODE REVIEW

File: doc/tags [hunk 1]
```diff
@@ -0,0 +1,26 @@
+buffer's	lspconfig.txt	/*buffer's*
```
Comment (clason) [inline doc/tags:1]: typo: `buffers`?

Comment (muniter) [reply doc/tags:1]: That is autogenerated by `:helptags` coming from this line https://github.com/neovim/nvim-lspconfig/blob/1c9c2120290d2fd6ea2cb55f954b5d108df0b628/doc/lspconfig.txt#L484 I'm guessing mj used that syntax to have some accent and it generates that on the tags.

Comment (clason) [reply doc/tags:1]: Gotcha. (I'd have written that as `*buffer*'s`, but to each their own ;))

Comment (mjlbach) [reply doc/tags:1]: Oh that tag shouldn't be there... Let's just change lspconfig.txt

File: doc/lspconfig.txt [hunk 2]
```diff
@@ -74,7 +74,7 @@ QUICKSTART                                                *lspconfig-quickstart*
     require'lspconfig'.clangd.setup{}
 <
 - create a new project, ensure it contains a root marker which matches the
-  server requirements specified in `server_configurations.md`.
+  server requirements specified in |lspconfig-servers|.
```
Comment (mjlbach) [inline doc/lspconfig.txt:77]: lspconfig-server-configurations maybe?

**Comment by mjlbach** [comment]
Great idea/improvement. Should we also mention this in the readme?

**Comment by muniter** [comment]
Went ahead and added the tag in some places in the readme, and `server_configurations.md` itself for people that discover it first on github. Also changed a few things, all ears in corrections and things you prefere not to :)

**Comment by muniter** [comment]
Also this is the logic that generates the tags in packer:
https://github.com/wbthomason/packer.nvim/blob/7f62848f3a92eac61ae61def5f59ddb5e2cc6823/lua/packer/plugin_utils.lua#L56-L73

Seems like as long as the tags modification time is later than the help file It won't regenerate the tags which should be the case here.

File: README.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ require'lspconfig'.rust_analyzer.setup{}
 EOF
 ```
 
-For a full list of servers, see [server_configurations.md](doc/server_configurations.md). This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be installed in less than a minute.
+For a full list of servers, see [server_configurations.md](doc/server_configurations.md) `:help lspconfig-server-configurations`. This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be installed in less than a minute.
```
Comment (mjlbach) [inline README.md:85]: "accessible via"

File: README.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ These features are not implemented in this repo, but in Neovim core. See `:help
     npm i -g pyright
     ```
 
-2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md)
+2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md) `:help lspconfig-server-configurations`
```
Comment (mjlbach) [inline README.md:44]: "accessible via"

File: doc/lspconfig.txt [hunk 5]
```diff
@@ -362,11 +365,12 @@ The three steps for adding and enabling a new server configuration are:
    lspconfig.foo_lsp.setup{}
 <
 ==============================================================================
-INDEX                                                          *lspconfig-index*
+CONFIGURATIONS                                 *lspconfig-server-configurations*
```
Comment (mjlbach) [inline doc/lspconfig.txt:368]: won't the tag from this conflict with the tag from the tagfile? Should we just delete this section?

File: doc/lspconfig.txt [hunk 6]
```diff
@@ -479,8 +477,8 @@ that require a client id can either leverage tab-completion or the info
 contained in `:LspInfo`:
 
 - `:LspStart <config_name>` launches the requested (configured) client, but only
-  if it successfully resolves a root directory. Defaults to all *configured* 
-  servers matching the *current* *buffer's* filetype.
+  if it successfully resolves a root directory. Note Defaults to all
```
Comment (mjlbach) [inline doc/lspconfig.txt:480]: I would remove note here

Comment (mjlbach) [reply doc/lspconfig.txt:480]: or add a comma/colon

**Comment by mjlbach** [comment]
I think before you had (hit K to to got server-configurations or something of the like) which was useful

**Comment by muniter** [comment]
The logic I'm thinking is: User needs to configure a servers does `:help lspconfig` looks for a configuration tag, first thing he sees is short paragraph telling him were they are. Sure can add the `K` thing.

**Comment by mjlbach** [comment]
Yeah totally, just want it to be clear to less experienced users how to jump to a tag.

**Comment by muniter** [comment]
Went with "type" since seems used a lot on `help`

File: doc/lspconfig.txt [hunk 7]
```diff
@@ -277,6 +279,9 @@ The global defaults for all servers can be overridden by extending the
 ==============================================================================
 SERVER CONFIGURATIONS                                 *lspconfig-configurations*
 
+See |lspconfig-server-configurations| by typing `K` over it for the complete
```
Comment (mjlbach) [inline doc/lspconfig.txt:282]: I would put this explanation in the first reference (intro) IMO

File: doc/lspconfig.txt [hunk 8]
```diff
@@ -74,10 +73,13 @@ QUICKSTART                                                *lspconfig-quickstart*
     require'lspconfig'.clangd.setup{}
 <
 - create a new project, ensure it contains a root marker which matches the
-  server requirements specified in `server_configurations.md`.
+  server requirements specified in |lspconfig-server-configurations|.
 
 - open a file within that project, such as `main.c`.
 
+- If you need more information about a server configuration read it's entry in
```
Comment (mjlbach) [inline doc/lspconfig.txt:80]: it's -> its

File: README.md [hunk 9]
```diff
@@ -41,7 +41,7 @@ These features are not implemented in this repo, but in Neovim core. See `:help
     npm i -g pyright
     ```
 
-2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md)
+2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md) accessible via `:help lspconfig-server-configurations`
```
Comment (clason) [inline README.md:44]:
```suggestion
2. Add the language server setup to your init.vim. The server name must match those found in the table of contents in [server_configurations.md](doc/server_configurations.md). This list is also accessible via `:help lspconfig-server-configurations`.
```

File: README.md [hunk 10]
```diff
@@ -82,7 +82,7 @@ require'lspconfig'.rust_analyzer.setup{}
 EOF
 ```
 
-For a full list of servers, see [server_configurations.md](doc/server_configurations.md). This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be installed in less than a minute.
+For a full list of servers, see [server_configurations.md](doc/server_configurations.md) accessible via `:help lspconfig-server-configurations`. This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup. Most language servers can be installed in less than a minute.
```
Comment (clason) [inline README.md:85]: no need to repeat here? Or simply "see [...](...) or `:help ...`.

Comment (clason) [inline README.md:85]:
```suggestion
For a full list of servers, see [server_configurations.md](doc/server_configurations.md) or `:help lspconfig-server-configurations`. This document contains installation instructions and additional, optional, customization suggestions for each language server. For some servers that are not on your system path (e.g., `jdtls`, `elixirls`), you will be required to manually add `cmd` as an entry in the table passed to `setup`. Most language servers can be installed in less than a minute.
```

File: doc/lspconfig.txt [hunk 11]
```diff
@@ -39,8 +38,8 @@ primary functionalities:
 `lspconfig` is not required to use the built-in client, it is only one front-end
 interface for when a language server specific plugin is not available.
 
-See |lspconfig-index| for the reference to the complete list of language
-servers.
+See |lspconfig-server-configurations| for the reference to the complete list
+of language servers.
```
Comment (clason) [inline doc/lspconfig.txt:42]:
```suggestion
See |lspconfig-server-configurations| for the complete list of language servers.
```

File: doc/lspconfig.txt [hunk 12]
```diff
@@ -74,10 +73,13 @@ QUICKSTART                                                *lspconfig-quickstart*
     require'lspconfig'.clangd.setup{}
 <
 - create a new project, ensure it contains a root marker which matches the
```
Comment (clason) [inline doc/lspconfig.txt:75]:
```suggestion
- create a new project, ensure that it contains a root marker which matches the
```

File: doc/lspconfig.txt [hunk 13]
```diff
@@ -218,7 +220,7 @@ passed overrides to `setup {}` are:
 
   The script that automatically generates `server_configurations.md` converts
   the `package.json` referenced in a server configuration source file
-  into a list of optional settings listed in `server_configurations.md`.
+  into a list of optional settings listed in |lspconfig-server-configurations|
```
Comment (clason) [inline doc/lspconfig.txt:223]:
```suggestion
  into a list of optional settings listed in |lspconfig-server-configurations|.
```

File: doc/lspconfig.txt [hunk 14]
```diff
@@ -74,10 +73,13 @@ QUICKSTART                                                *lspconfig-quickstart*
     require'lspconfig'.clangd.setup{}
 <
 - create a new project, ensure it contains a root marker which matches the
-  server requirements specified in `server_configurations.md`.
+  server requirements specified in |lspconfig-server-configurations|.
 
 - open a file within that project, such as `main.c`.
 
+- If you need more information about a server configuration read it's entry in
+  |lspconfig-server-configurations|
```
Comment (clason) [inline doc/lspconfig.txt:81]:
```suggestion
- If you need more information about a server configuration, read the corresponding
  entry in |lspconfig-server-configurations|.
```

**Review by clason — APPROVED** [review]

**Comment by mjlbach** [comment]
Couldn't push to the branch so committed clason's changes in https://github.com/neovim/nvim-lspconfig/pull/1497

**Pull request closed without merging**